backend/post-service/src/main/java/com/speakapp/postservice/services/PostService.java

```diff
@@ -48,12 +51,43 @@ public PostGetDTO createPost(PostCreateDTO postCreateDTO, UUID userId) {
 
         return postMapper.toGetDTO(savedPost,
                 author,
-                commentGetDTOS,
                 reactionsGetDTO,
                 null
         );
     }
 
+    public LatestUserPostsPageGetDTO getUsersLatestPosts(int pageNumber, int pageSize, UUID userIdOfProfileOwner, UUID userId){
+        Pageable page = PageRequest.of(pageNumber, pageSize);
+        Page<Post> userPostsPage = postRepository.findAllByUserIdOrderByCreatedAtDesc(userIdOfProfileOwner, page);
+
+        List<PostGetDTO> userPostGetDTOS = new ArrayList<>();
```

Comment:
to co tutaj robisz to konwersja page -> list -> page, a masz coś takiego jak map dla interfejsu Pageable, zerknij sobie na artykuł: https://stackoverflow.com/questions/48867377/is-there-a-auto-mapping-for-spring-pageable-sort-with-mapstruct

W zasadzie mogłoby to zostać ale znając ciebie to pewnie byś wolał to zrobić w sprytniejszy sposób, w zasadzie można tu ominąć całkiem pętle i mieć dosłownie: userPostsPage.map() - chętnie mogę to sam napisać i pokazać o co mi cho

Reply:
Przeanalizuję, w razie wątpliwości dam znać.